compiler/luci/pass/src/FuseBCQPass.cpp

```diff
@@ -107,7 +122,13 @@ class BCQConverter final
   bool do_w_x(luci::CircleConst *node)
   {
     const auto prefix = node_name_prefix(node->name());
-    return _do_w_x[prefix]->at<loco::DataType::BOOL>(0);
+
+    if (_do_w_x[prefix]->dtype() == loco::DataType::S32)
+      return _do_w_x[prefix]->at<loco::DataType::S32>(0) == 1;
+    else if (_do_w_x[prefix]->dtype() == loco::DataType::BOOL)
+      return _do_w_x[prefix]->at<loco::DataType::BOOL>(0);
+    else
+      throw std::runtime_error("do_w_x should be int or bool");
```

Comment:
stop the program is not good. can you just return false?

Reply:
I agree that stopping the program is not good :)

However, if `false` is returned, it can cause unintended behavior because we don't know what was original intention.
For example, if `dtype` is `FLOAT32` and the value was 0.5, it is not `true` nor `false`. Thus I thought that user should be notified BCQ cannot be applied.

I can change the codes to return `false` at `run()` function when the information is invalid, but if then, should I apply the diff to 1.7.0 either? Because this code is included at 1.7.0 too :(

Reply:
For our optimization pass, return false is do nothing. I wish this will apply here also.

> should I apply the diff to 1.7.0 either?

yes please.

Reply:
> For our optimization pass, return false is do nothing

I said about `bool do_w_x(luci::CircleConst *node)`, not optimization pass :)
In short, if `do_w_x` returns `false`, it means that we would ignore unexpected behavior.

Anyway, I will change related codes.